src/Analyzer.Core/TemplateAnalyzer.cs

```diff
@@ -130,28 +173,76 @@ public IEnumerable<IEvaluation> AnalyzeTemplate(string template, string paramete
+                dynamic processedTemplateResourcesWithLineNumbers = templateContext.OriginalTemplate["resources"];
 
-                // For each rule we don't want to report the same line more than once
-                // This is a temporal fix
-                var evalsToValidate = new List<IEvaluation>();
-                var evalsToNotValidate = new List<IEvaluation>();
-                foreach (var eval in evaluations)
+                for (int i = 0; i < processedTemplateResources.Count; i++)
                 {
-                    if (!eval.Passed && eval.Result != null)
-                    {
-                        evalsToValidate.Add(eval);
-                    }
-                    else
+                    dynamic currentProcessedResource = processedTemplateResources[i];
+                    dynamic currentProcessedResourceWithLineNumbers = processedTemplateResourcesWithLineNumbers[i];
+
+                    if (currentProcessedResource.type == "Microsoft.Resources/deployments")
                     {
-                        evalsToNotValidate.Add(eval);
+                        dynamic nestedTemplateWithLineNumbers = currentProcessedResourceWithLineNumbers.properties.template;
+                        dynamic populatedNestedTemplate = JsonConvert.DeserializeObject(JsonConvert.SerializeObject(nestedTemplateWithLineNumbers));
+                        // get the offset
+                        int nextLineNumberOffset = (nestedTemplateWithLineNumbers as IJsonLineInfo).LineNumber + LineNumberOffset - 1; // off by one
+                        // check whether scope is set to inner or outer
+                        var scope = currentProcessedResourceWithLineNumbers.properties.expressionEvaluationOptions?.scope;
+
+                        int startOfTemplate = (nestedTemplateWithLineNumbers as IJsonLineInfo).LineNumber;
+                        string jsonNestedTemplate = ExtractNestedTemplate(template, startOfTemplate);
+                        IEnumerable<IEvaluation> result;
+
+                        if (scope == null)
+                        {
+                            scope = "outer";
+                        }
+                        if (scope == "inner")
+                        {
+                            // Pass parameters, variables and functions to child template
+                            JToken passedParameters = currentProcessedResource.properties.parameters;
+                            JToken passedVariables = currentProcessedResource.properties.variables;
+                            JToken passedFunctions = currentProcessedResource.properties.functions;
+
+                            // merge 
+                            populatedNestedTemplate.variables?.Merge(passedVariables);
+                            populatedNestedTemplate.functions?.Merge(passedFunctions);
+
+                            dynamic currentPassedParameter = passedParameters?.First;
```

Comment:
The templateProcesser expects the field "defaultValue" and not value. However, at this point, the processed template parameters have the field 'value' which holds the actual values of parameters. 
Since I need to make the nested template valid for the next function call, I parse through all the parameters, replacing the 'value' field with 'defaultValue' so it can be a valid template in the next recursive call